lib/Parser/UOP/factorial.pm

```diff
@@ -25,6 +25,7 @@ sub _eval {
   my $n = shift; my $f = 1;
   $self->Error("Factorial can only be taken of (non-negative) integers")
     unless $n =~ m/^\d+$/;
+  return Inf if $n > 170;
```

Comment:
I'm wondering if this should produce a MathObject Infinity rather than a Perl `Inf`.  (My comment on the original discussion used `Inf` to make it work like the other ones in Alex's tests.)

I get error messages about `Inf` being a bareword for this line.  I think it may need to be

```suggestion
  return $self->Package("Infinity")->new() if $n > 170;
```

Reply:
I tried the proposed change now.

Even with that change putting `171!` inside a n input box for a Formula results in the error message `Can't convert an Infinity to a constant`.

`Compute( "171!" );` also triggers that error for either option of the return value for n>170 (and also when the prior version of the code is used).

As such, it seems to me that keep the return value as a Perl infinity (which it would have been in the past) is a more certain way of maintaining backwards compatibility, and saves on any overhead of creating an object here when it has no purpose in the end.

I do see a warning about the bareword from `perl -C lib/Parser/UOP/factorial.pm` but it does not seem to interfere when used inside pg.

However, if you still think the change is advisable, I will make it.

Reply:
> Even with that change putting 171! inside a n input box for a Formula results in the error message Can't convert an Infinity to a constant.

I am not able to reproduce that.  It works correctly for me (reporting an entered value of `infinity` and that the value is incorrect.  No error message is produced.

Reply:
I upgraded my version of PG. The Docker container had a somewhat old version. Now it works properly. Sorry.

Reply:
Note: Peter's PR had the recommended change in it.

Reply:
That is now merged, so `$self->Package("Infinity")->new()` is not the return value of n>170.